probe-rs/src/config/target.rs

```diff
@@ -0,0 +1,63 @@
+use super::chip::Chip;
+use super::chip_family::ChipFamily;
+use super::flash_algorithm::{FlashAlgorithm, RawFlashAlgorithm};
+use super::memory::{FlashRegion, MemoryRegion, RamRegion};
+use super::registry::TargetIdentifier;
+use crate::target::Core;
+use jep106::JEP106Code;
+
+/// This describes a complete target with a fixed chip model and variant.
+#[derive(Debug, Clone)]
+pub struct Target {
+    /// The complete identifier of the target.
+    pub identifier: TargetIdentifier,
+    /// The JEP106 code of the manufacturer.
+    pub manufacturer: Option<JEP106Code>,
+    /// The `PART` register of the chip.
+    /// This value can be determined via the `cli info command`.
+    pub part: Option<u16>,
+    /// The name of the flash algorithm.
+    pub flash_algorithm: Option<FlashAlgorithm>,
+    /// The core type.
+    pub core: Box<dyn Core>,
+    /// The memory map of the target.
+    pub memory_map: Vec<MemoryRegion>,
+}
+
+pub type TargetParseError = serde_yaml::Error;
+
+impl
+    From<(
+        &ChipFamily,
```

Comment:
I think a normal function with parameters, would be better here, I find large tuples hard to understand.

Reply:
Dunno what was in me ... fixed.